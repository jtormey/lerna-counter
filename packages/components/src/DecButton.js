import React from 'react'
import Button from './Button'

const DecButton = ({ onDec }) => (
  <Button title='decrement' onClick={onDec} />
)

export default DecButton
