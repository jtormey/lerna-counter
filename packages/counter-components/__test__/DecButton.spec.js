import React from 'react'
import DecButton from '../src/DecButton'
import { render, mount } from 'enzyme'

describe('DecButton', () => {
  it('should render a button', () => {
    let wrapper = render(
      <DecButton />
    )
    expect(wrapper.toString()).toMatchSnapshot()
  })

  it('should trigger an increment', () => {
    let mockFn = jest.fn()
    let wrapper = mount(
      <DecButton onDec={mockFn} />
    )
    wrapper.find('button').simulate('click')
    expect(mockFn.mock.calls.length).toEqual(1)
  })
})
