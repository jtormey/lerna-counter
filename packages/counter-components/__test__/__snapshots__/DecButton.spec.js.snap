// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecButton should render a button 1`] = `"<button>decrement</button>"`;
